### ISSUE TITLE
ZH03B error reading stream

### ISSUE THREAD

**Issue by Sholahudin** [ShaggyDog18/SD_ZH03B#4]
I have trouble about reading the stream by ZH03B. Could please help me?

**Comment by ShaggyDog18** [comment]
How can I help you without knowing anything about trouble you facing? Please, describe.
Try to compile and run an example: 
- check up UART settings. 
- check up the sensor ZH03B connection to microcontroller (RX to TX, TX to RX)... 
- check up the mode in which you are running ZH03B. To learn about modes, please, refer to datasheet.

**Comment by Sholahudin** [comment]
I am really sorry, the previous problem caused by my jumper cable (bad).

Now, i am having this on my serial monitor. The mode is interchange automatic between Q&A and IU mode. Which one is the best mode ? And how to fix that? Thank you.
![Screen Shot 2024-03-05 at 05 36 21](https://github.com/ShaggyDog18/SD_ZH03B/assets/79613788/73516fc9-ccd5-4c2f-9c16-8c37a50fbcc0)

**Comment by Sholahudin** [comment]
Is that up to me (means that I can comment the code for mode that i don't need)?

**Comment by Sholahudin** [comment]
The value is in ug/m3 right? and why i getting error conitinuously after testing it with spraying perfume (indirect little bit far)? And back to normal after push the UNO reset button.

![Screen Shot 2024-03-05 at 06 16 05](https://github.com/ShaggyDog18/SD_ZH03B/assets/79613788/154cdea9-24e2-4447-ba47-8d6c1ff147df)

**Comment by Sholahudin** [comment]
And, now, without any dust test, my sensor is getting error after several minutes reading.

![Screen Shot 2024-03-05 at 06 29 56](https://github.com/ShaggyDog18/SD_ZH03B/assets/79613788/f358962b-4fa3-43d6-aa74-2773d4599f15)

**Comment by Sholahudin** [comment]
This is my sensor and Arduino UNO connection.

![IMG_4024](https://github.com/ShaggyDog18/SD_ZH03B/assets/79613788/b8dfb914-0af1-40af-afeb-4546a6a1a514)

**Comment by ShaggyDog18** [comment]
Right, the example is comprehensive includes/tests all the features of the sensor. You may simplify it by removing almost everything: sleep() methods as well as switching to other mode, i.e. you loop may look like:
```
void loop () {
  Serial.println( "Normal default IU Mode. reading every 5 sec" );
  for( int i=0; i<4; i++ ){
    readSensorData();
    delay(5000);
  }
}
```
Right, select the operation mode you would like to use to communicate with your sensor: UI or QA (described in details at the home page). I cannot judge what is better in your particular case. It depends on your  application and requirements.
Use UI for testing, it it simple and available by default.

**Comment by ShaggyDog18** [comment]
In case you still have issues reading the stream, try to clean RX buffer by the following function:
```
void RX_flush(void){
  delay(100);   
  while(  _serial.available() ) {
     _serial.read();
  } 
}
```
Try reading every second (it is reading every 5 seconds in the example), if errors, clean RX after every sensor read.
If still errors, try using Q&A operations mode. This mode could be more stable: you request the sensor, and receive the data stream in the response. So, RX should not override (it potentially could in UI mode).
